### DISCUSSION TITLE
Inquire about the ability to support native speed for Linux instead of using a VM

### DISCUSSION THREAD

**Discussion by foodmaher** [microsoft/WSL#10806] [General]
Now, I have some apps that are only available for Linux, such as Easypulse, and my laptop is too slow to deal with WSLG for Linux applications on Windows 11. So, my request is, **If we have full support for the Linux kernel on Windows, why are we not able to get the native speed of it instead of using VM for Linux, which makes the experience worse (as you know)?**

I'm not a developer for any operating system, but I found that crazy because we have Linux kernel on Windows as an additional feature. However, we still use a VM machine to run Linux through WSL2 (which also can make Android apps work more smoothly because Linux has Waydroid work as a container to provide native speed for Android apps instead of WSA (I don't know why Microsoft still use VM solution instead of use containers as Waydroid for example)).

I apologise for my request structure; I have not followed the official format. 
I hope these features will be added with WSL3 or the next version of Windows, which will kill all other operating systems.

**Comment by trungnt2910** [comment]
Not from Microsoft, but we can all guess:

- **License issue**: Linux is under the GPLv2, so modifying it and putting it into a kernel driver like `lxcore` for WSL1 would open up opportunities for people to argue that the NT kernel itself would be infected by that license. Microsoft might want to avoid this and shield themselves using a VM layer.
- **Compatibility issues**: To make such a version of Linux to work well as a Pico driver, one would need to somehow extend the Linux kernel to be able to map its processes to NT Pico processes, access Windows files through a cooperative FS driver, forward UNIX/INET sockets, and much more. This does require some work (but not impossible).
- **Microsoft being lazy**: This is probably the main reason. If you notice Microsoft's trends in the 2019-2020 period, you can see that they had been killing a lot of core components of the Windows 10 ecosystem, including [Windows 10 Mobile](https://support.microsoft.com/en-us/windows/windows-10-mobile-end-of-support-faq-8c2dd1cf-a571-00f0-0881-bb83926d05c5#:~:text=Only%20device%20models%20that%20are%20eligible%20for%20Windows,Mobile%20availability%20per%20device%20model%20for%20more%20information.), [UWP Microsoft Edge](https://support.microsoft.com/en-us/microsoft-edge/what-is-microsoft-edge-legacy-3e779e55-4c55-08e6-ecc8-2333768c0fb0), and of course, WSL1. They are pouring their resources into boring yet profitable stuff such as AI and Azure cloud. Only the projects supporting their goals (like .NET or Linux container images) are focused on.

If you want such "native" support, you can try creating your own drivers, [something like this](https://github.com/trungnt2910/lxmonika/tree/master/mxss). Be prepared to wade through a lot of undocumented NT kernel features and interfaces though.
